templates/decays.cpp.in

```diff
@@ -0,0 +1,306 @@
+      const context_base& context,
+      const typename field_indices<H>::type& in_idx,
+      const typename field_indices<A>::type& out1_idx,
+      const typename field_indices<A>::type& out2_idx) const
+{
+   const auto amp = calculate_amplitude<H, A, A>(context, in_idx, out1_idx, out2_idx);
+   const double mH = context.physical_mass<H>(in_idx);
+   constexpr double ps {1./(8.*Pi)};
+   constexpr double ps_symmetry {1./2.};
+   const double color_gen_sqr = squared_color_generator<H, A, A>();
+   const double flux = 0.5/mH;
+   const auto res = flux * color_gen_sqr * ps * ps_symmetry * amp.square();
+
+   if (res < 0) {
+      throw std::runtime_error("Error in H->AA. Partial width < 0");
+   } else {
+      return res;
+   }
+}
+
+template <>
+double CLASSNAME::get_partial_width<H, A, Z>(
+      const context_base& context,
+      const typename field_indices<H>::type& in_idx,
+      const typename field_indices<A>::type& out1_idx,
+      const typename field_indices<Z>::type& out2_idx) const
+{
+   const double mH = context.physical_mass<H>(in_idx);
+   const double mA = context.physical_mass<A>(out1_idx);
+   const double mZ = context.physical_mass<Z>(out2_idx);
+
+   const double ps {1./(8.*Pi) * sqrt(KallenLambda(1., Sqr(mA/mH), Sqr(mZ/mH)))};
+   constexpr double ps_symmetry {1.};
+   const double color_gen_sqr = squared_color_generator<H, A, Z>();
+   const double flux = 0.5/mH;
+
+   const auto amp = calculate_amplitude<H, A, Z>(context, in_idx, out1_idx, out2_idx);
+
+   const auto res = flux * color_gen_sqr * ps * ps_symmetry * amp.square();
+
+   if (res < 0) {
+      throw std::runtime_error("Error in H->AZ. Partial width < 0");
+   } else {
+      return res;
+   }
+}
+
+template <>
+double CLASSNAME::get_partial_width<H, G, G>(
+      const context_base& context,
+      const typename field_indices<H>::type& in_idx,
+      const typename field_indices<G>::type& out1_idx,
+      const typename field_indices<G>::type& out2_idx) const
+{
+   const auto amp = calculate_amplitude<H, G, G>(context, in_idx, out1_idx, out2_idx);
+   const double mH = context.physical_mass<H>(in_idx);
+   constexpr double ps {1./(8.*Pi)};
+   constexpr double ps_symmetry {1./2.};
+   const double color_fact = squared_color_generator<H, G, G>();
+   const double flux = 0.5/mH;
+
+   double result = flux * color_fact * ps * ps_symmetry * amp.square();
+
+   // higher order corrections to CP-even Higgs
+   if (!info::is_CP_violating_Higgs_sector) {
+      // higher order QCD corrections
+
+      // number of active light flavours
+      unsigned int Nf;
+      const double mtpole {qedqcd.displayPoleMt()};
+      if (mH > 5 && mH < mtpole) {
+         Nf = 5;
+      } else if (mH > mtpole) {
+         Nf = 6;
+      } else {
+         throw;
+      }
+
+      // NLO, NNLO - eq. 11 of hep-ph/9705240
+      const double deltaNLO {95./4. - 7./6.*Nf};
+
+      const double mtpole2 {Sqr(mtpole)};
+      using boost::math::zeta;
```

Comment:
Please remove boost's zeta function (and the corresponding header) and replace it by fixed numerical values that are locally defined:
~~~
const double zeta2 = 1.6449340668482264;
const double zeta3 = 1.2020569031595943;
~~~
Reason: This is guaranteed to be fastest and avoids inclusion of the boost header (and thus reduces compile time).

Reply:
Ok. Let me just explain why I was usually doing stuff like that. If I see a line like
```C++
const double zeta2 = 1.6449340668482264;
```
I cannot say if it's wrong or not. As I have no idea what is zeta(2). If someone ever by accident flips a digit I will never find it. While something like
```C++
const double zeta2 = zeta(2);
```
I can check. I understand the speed argument but I think we're sacrificing readability in favour of probably miniscule speed gain.

Reply:
Hmm, I see what you mean. Maybe we can both be happy by moving the definition of these constants to some header, (like `wrappers.hpp`) and add a corresponding unit test that ensures that `zeta2 == zeta(2)`.

Anyway, I see no point in calculating `zeta(2)` each time. It would be like calculating `pi` each time using some function. `pi` will never change, as will `zeta2`. :)